internal/code-generator/generator/heuristics.go

```diff
@@ -72,24 +70,17 @@ func loadRule(namedPatterns map[string]StringArray, rule *Rule) *LanguagePattern
 		}
 		result = &LanguagePattern{"And", rule.Languages, "", subPatterns}
 	} else if len(rule.Pattern) != 0 { // OrPattern
-		conjunction := strings.Join(rule.Pattern, orPipe)
-		pattern := convertToValidRegexp(conjunction)
+		pattern := strings.Join(rule.Pattern, orPipe)
 		result = &LanguagePattern{"Or", rule.Languages, pattern, nil}
 	} else if rule.NegativePattern != "" { // NotPattern
-		pattern := convertToValidRegexp(rule.NegativePattern)
-		result = &LanguagePattern{"Not", rule.Languages, pattern, nil}
+		result = &LanguagePattern{"Not", rule.Languages, rule.NegativePattern, nil}
 	} else if rule.NamedPattern != "" { // Named OrPattern
-		conjunction := strings.Join(namedPatterns[rule.NamedPattern], orPipe)
-		pattern := convertToValidRegexp(conjunction)
+		pattern := strings.Join(namedPatterns[rule.NamedPattern], orPipe)
 		result = &LanguagePattern{"Or", rule.Languages, pattern, nil}
 	} else { // AlwaysPattern
 		result = &LanguagePattern{"Always", rule.Languages, "", nil}
 	}
 
-	if isUnsupportedRegexpSyntax(result.Pattern) {
```

Comment:
Did I got it right that in case of RE2, this was moved from a build-time to the run-time check?  

If so, I wonder
 * did you notice any quantifiable runtime performance cost of these checks?
 * does the additional tests guarantee that RE2 does not "surprise" users accidentally at run-time, e.g not or stop matching (as default behaviour)?

Reply:
It has no penalty since the regexp is really slow. You can run the benchmark and see that is no difference compared to the previous version. 

The validation is done at runtime using the same [algortim](https://github.com/go-enry/go-enry/pull/65/files#diff-22f255a3c764d1661c6db32070d047ef71fecf89dd616e72200d0a11059d006cR37-R49) that we used to run in the code generation, so the people will have exactly the same behavior, in case you are using the regexp package. In the case of using go-oniguruma, you will find that results are now more accurate to go-linguist. 

The main problem is that due to limitations with the regular expression with are falling on the detection of many languages, if we run the full fixtures list against go-enry you will find that many fixtures are failing. With this change at least we provide the option to the user to be more accurate in the language detection.

This code was already running at gitsight for some months and works perfectly.

Reply:
Thank you for the explanation, it makes perfect sense now.